optuna/integration/allennlp.py

```diff
@@ -9,17 +10,127 @@
+    # On the other hand, `EpochCallback` defined here is not `Registerable`,
+    # which causes a mypy checking feailure.
+    class EpochCallback:  # type: ignore
+        """Stub for EpochCallback.
+
+        This class is introduced for documentation CI.
+
+        """
+
+        @classmethod
+        def register(cls: Any, *args: Any, **kwargs: Any) -> Callable:
+            def wrapper(subclass: Any, *args: Any, **kwargs: Any) -> None:
+                return subclass
+
+            return wrapper
+
+
+PPID = os.getppid()
+
+
+def _create_pruner() -> Optional[optuna.pruners.BasePruner]:
+    pruner_class = os.getenv("{}_OPTUNA_ALLENNLP_PRUNER_CLASS".format(PPID))
+    if pruner_class is None:
+        return None
+
+    pruner_params = _get_environment_variables_for_pruner()
+
+    pruner = None  # type: Any
+    if pruner_class == "HyperbandPruner":
+        pruner = optuna.pruners.HyperbandPruner
+    elif pruner_class == "MedianPruner":
+        pruner = optuna.pruners.MedianPruner
+    elif pruner_class == "PercentilePruner":
+        pruner = optuna.pruners.PercentilePruner
+    elif pruner_class == "ThresholdPruner":
+        pruner = optuna.pruners.ThresholdPruner
+    elif pruner_class == "SuccessiveHalvingPruner":
+        pruner = optuna.pruners.SuccessiveHalvingPruner
+    elif pruner_class == "NopPruner":
+        pruner = optuna.pruners.NopPruner
+
+    if pruner is None:
+        return None
+
+    return pruner(**pruner_params)
+
+
+def _get_environment_variables_for_trial() -> Dict[str, Optional[str]]:
+    return {
+        "study_name": os.getenv("{}_OPTUNA_ALLENNLP_STUDY_NAME".format(PPID)),
+        "trial_id": os.getenv("{}_OPTUNA_ALLENNLP_TRIAL_ID".format(PPID)),
+        "storage": os.getenv("{}_OPTUNA_ALLENNLP_STORAGE_NAME".format(PPID)),
+        "monitor": os.getenv("{}_OPTUNA_ALLENNLP_MONITOR".format(PPID)),
+    }
+
+
+def _get_environment_variables_for_pruner() -> Dict[str, Optional[str]]:
+    keys = os.getenv("{}_OPTUNA_ALLENNLP_PRUNER_KEYS".format(PPID))
+    if keys is None:
+        return {}
+
+    kwargs = {}
+    for key in keys.split(","):
+        value = os.getenv("{}_OPTUNA_ALLENNLP_{}".format(PPID, key))  # type: Any
+
+        try:
+            value = int(value)
+        except ValueError:
+            try:
+                value = float(value)
+            except ValueError:
+                pass
+
+        kwargs[key] = value
+
+    return kwargs
+
+
+def _export_pruner_config(trial: optuna.Trial) -> Dict[str, Any]:
+    pruner = trial.study.pruner
+    kwargs = {}  # type: Dict[str, Any]
+
+    if isinstance(pruner, optuna.pruners.HyperbandPruner):
+        kwargs["min_resource"] = pruner._min_resource
+        kwargs["max_resource"] = pruner._max_resource
+        kwargs["reduction_factor"] = pruner._reduction_factor
+    elif isinstance(pruner, optuna.pruners.MedianPruner) or isinstance(
+        pruner, optuna.pruners.PercentilePruner
+    ):
```

Comment:
Does this work?
```suggestion
    elif isinstance(pruner, optuna.pruners.PercentilePruner):
```

Ref: https://github.com/optuna/optuna/blob/fe1e3e54f961897b871daa7fe23d92c4e1eaa408/optuna/pruners/_median.py#L4